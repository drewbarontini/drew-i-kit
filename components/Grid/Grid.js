import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../lib/styleUtils';

const Grid = styled.div`
  display: grid;

  ${setProp({
    prop: 'alignContent',
    cssProp: 'align-content',
  })};
  ${setProp({
    prop: 'alignItems',
    cssProp: 'align-items',
  })};
  ${setProp({
    prop: 'areas',
    cssProp: 'grid-template-areas',
  })};
  ${setProp({
    prop: 'autoColumns',
    cssProp: 'grid-auto-columns',
  })};
  ${setProp({
    prop: 'autoFlow',
    cssProp: 'grid-auto-flow',
  })};
  ${setProp({
    prop: 'autoRows',
    cssProp: 'grid-auto-rows',
  })};
  ${setProp({
    prop: 'columns',
    cssProp: 'grid-template-columns',
  })};
  ${setProp({
    prop: 'columnGap',
    cssProp: 'grid-column-gap',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'justifyContent',
    cssProp: 'justify-content',
  })};
  ${setProp({
    prop: 'gap',
    cssProp: 'grid-gap',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'grid',
  })};
  ${setProp({
    prop: 'placeContent',
    cssProp: 'place-content',
  })};
  ${setProp({
    prop: 'placeItems',
    cssProp: 'place-items',
  })};
  ${setProp({
    prop: 'rows',
    cssProp: 'grid-template-rows',
  })};
  ${setProp({
    prop: 'rowGap',
    cssProp: 'grid-row-gap',
    themeKey: 'spacing',
  })};
  ${setProp({
    prop: 'template',
    cssProp: 'grid-template',
  })};
`;

Grid.propTypes = {
  alignContent: PropTypes.string,
  alignItems: PropTypes.string,
  autoColumns: PropTypes.string,
  autoFlow: PropTypes.string,
  autoRows: PropTypes.string,
  areas: PropTypes.string,
  columns: PropTypes.string,
  columnGap: PropTypes.string,
  gap: PropTypes.string,
  grid: PropTypes.string,
  justifyContent: PropTypes.string,
  placeContent: PropTypes.string,
  placeItems: PropTypes.string,
  rows: PropTypes.string,
  rowGap: PropTypes.string,
  template: PropTypes.string,
};

Grid.displayName = 'Grid';

export default Grid;
