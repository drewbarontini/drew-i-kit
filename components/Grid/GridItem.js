import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../lib/styleUtils';

const GridItem = styled.div`
  ${setProp({
    prop: 'alignSelf',
    cssProp: 'align-self',
  })};
  ${setProp({
    prop: 'area',
    cssProp: 'grid-area',
  })};
  ${setProp({
    prop: 'column',
    cssProp: 'grid-column',
  })};
  ${setProp({
    prop: 'columnEnd',
    cssProp: 'grid-column-end',
  })};
  ${setProp({
    prop: 'columnStart',
    cssProp: 'grid-column-start',
  })};
  ${setProp({
    prop: 'justifySelf',
    cssProp: 'justify-self',
  })};
  ${setProp({
    prop: 'placeSelf',
    cssProp: 'place-self',
  })};
  ${setProp({
    prop: 'row',
    cssProp: 'grid-row',
  })};
  ${setProp({
    prop: 'rowEnd',
    cssProp: 'grid-row-end',
  })};
  ${setProp({
    prop: 'rowEnd',
    cssProp: 'grid-row-end',
  })};
`;

GridItem.propTypes = {
  alignSelf: PropTypes.string,
  area: PropTypes.string,
  column: PropTypes.string,
  columnEnd: PropTypes.string,
  columnStart: PropTypes.string,
  justifySelf: PropTypes.string,
  placeSelf: PropTypes.string,
  row: PropTypes.string,
  rowEnd: PropTypes.string,
  rowStart: PropTypes.string,
};

GridItem.displayName = 'GridItem';

export default GridItem;
