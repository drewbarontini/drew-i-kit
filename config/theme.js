const theme = {
  borders: {
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  spacing: {
    base: '16px',
    xs: '5px',
    s: '10px',
    m: '12px',
    l: '32px',
    xl: '60px',
  },
};

export default theme;
