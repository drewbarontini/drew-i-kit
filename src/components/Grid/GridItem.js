import PropTypes from 'prop-types';
import styled from 'styled-components';

import { setProp } from '../../lib/props';
import { responsive } from '../../lib/responsive';

const GridItem = styled.div`
  ${setProp({
    prop: 'alignSelf',
  })};
  ${setProp({
    prop: 'area',
    cssProp: 'grid-area',
  })};
  ${setProp({
    prop: 'column',
    cssProp: 'grid-column',
  })};
  ${setProp({
    prop: 'columnEnd',
    cssProp: 'grid-column-end',
  })};
  ${setProp({
    prop: 'columnStart',
    cssProp: 'grid-column-start',
  })};
  ${setProp({
    prop: 'justifySelf',
  })};
  ${setProp({
    prop: 'placeSelf',
  })};
  ${setProp({
    prop: 'row',
    cssProp: 'grid-row',
  })};
  ${setProp({
    prop: 'rowEnd',
    cssProp: 'grid-row-end',
  })};
  ${setProp({
    prop: 'rowEnd',
    cssProp: 'grid-row-end',
  })};

  ${responsive};
`;

GridItem.propTypes = {
  alignSelf: PropTypes.string,
  area: PropTypes.string,
  column: PropTypes.string,
  columnEnd: PropTypes.string,
  columnStart: PropTypes.string,
  justifySelf: PropTypes.string,
  placeSelf: PropTypes.string,
  row: PropTypes.string,
  rowEnd: PropTypes.string,
  rowStart: PropTypes.string,
};

GridItem.displayName = 'GridItem';

export default GridItem;
