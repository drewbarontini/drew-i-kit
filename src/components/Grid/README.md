# Grid

### Usage

```jsx
import { Grid, GridItem } from 'drew-i-kit';

<Grid>
  <GridItem>
    <p>Hello</p>
  </GridItem>
  <GridItem>
    <p>Grid</p>
  </GridItem>
</Grid>
```

```jsx
import { A } from 'drew-i-kit';

<Grid
  gap="base"
  columns="repeat(9, 1fr)"
  autoRows="minmax(100px, auto)"
  areas="
  'hd hd hd hd   hd   hd   hd   hd   hd'
  'sd sd sd mn   mn   mn   mn   mn   mn'
  'ft ft ft ft   ft   ft   ft   ft   ft'
"
>
  <GridItem area="hd">
    <p>Header</p>
  </GridItem>
  <GridItem area="sd">
    <p>Sidebar</p>
  </GridItem>
  <GridItem area="mn">
    <p>Main</p>
  </GridItem>
  <GridItem area="ft">
    <p>Footer</p>
  </GridItem>
</Grid>
```

### Props

#### `Grid`
- `alignContent` (**string**, optional)
- `alignItems` (**string**, optional)
- `autoColumns` (**string**, optional)
- `autoFlow` (**string**, optional)
- `autoRows` (**string**, optional)
- `areas` (**string**, optional)
- `columns` (**string**, optional)
- `columnGap` (**string**, optional)
- `gap` (**string**, optional)
- `grid` (**string**, optional)
- `justifyContent` (**string**, optional)
- `placeContent` (**string**, optional)
- `placeItems` (**string**, optional)
- `rows` (**string**, optional)
- `rowGap` (**string**, optional)
- `template` (**string**, optional)

#### `GridItem`
- `alignSelf` (**string**, optional)
- `area` (**string**, optional)
- `column` (**string**, optional)
- `columnEnd` (**string**, optional)
- `columnStart` (**string**, optional)
- `justifySelf` (**string**, optional)
- `placeSelf` (**string**, optional)
- `row` (**string**, optional)
- `rowEnd` (**string**, optional)
- `rowStart` (**string**, optional)
