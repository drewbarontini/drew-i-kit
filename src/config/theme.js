const theme = {
  borders: {
    sizes: {
      base: '1px',
    },
    radius: {
      base: '3px',
    },
  },
  colors: {
    bg: '#fff',
    fg: '#272727',
    primary: '#3576a7',
    subdued: '#cbcbcb',
    error: '#fe5f55',
    success: '#8acb88',
    warning: '#ffc527',
    white: '#fff',
    black: '#272727',
  },
  fonts: {
    family: {
      base:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
      heading:
        "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    },
    lineHeight: {
      base: '1.5',
      heading: '1.2',
    },
    sizes: {
      base: '14px',
      xxs: '8px',
      xs: '10px',
      s: '12px',
      l: '16px',
      xl: '24px',
      h1: '26px',
    },
    weights: {
      light: 200,
      normal: 400,
      semiBold: 600,
      bold: 800,
    },
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
