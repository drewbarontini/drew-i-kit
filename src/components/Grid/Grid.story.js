import React from 'react';
import { storiesOf } from '@storybook/react';

import Box from '../Box';
import Flex from '../Flex';
import Grid, { GridItem } from '.';
import Text from '../Text';

storiesOf('Grid', module).add('default', () => (
  <Grid
    gap="base"
    columns="repeat(9, 1fr)"
    autoRows="minmax(100px, auto)"
    areas="
    'hd hd hd hd   hd   hd   hd   hd   hd'
    'sd sd sd mn   mn   mn   mn   mn   mn'
    'ft ft ft ft   ft   ft   ft   ft   ft'
  "
  >
    <GridItem area="hd">
      <Box height="100%" backgroundColor="primary.base" padding="base">
        <Flex center height="100%">
          <Text color="bg" fontWeight="bold" uppercase>
            Header
          </Text>
        </Flex>
      </Box>
    </GridItem>
    <GridItem area="sd">
      <Box height="100%" backgroundColor="primary.base" padding="base">
        <Flex center height="100%">
          <Text color="bg" fontWeight="bold" uppercase>
            Sidebar
          </Text>
        </Flex>
      </Box>
    </GridItem>
    <GridItem area="mn">
      <Box height="100%" backgroundColor="primary.base" padding="base">
        <Flex center height="100%">
          <Text color="bg" fontWeight="bold" uppercase>
            Main
          </Text>
        </Flex>
      </Box>
    </GridItem>
    <GridItem area="ft">
      <Box height="100%" backgroundColor="primary.base" padding="base">
        <Flex center height="100%">
          <Text color="bg" fontWeight="bold" uppercase>
            Footer
          </Text>
        </Flex>
      </Box>
    </GridItem>
  </Grid>
));
