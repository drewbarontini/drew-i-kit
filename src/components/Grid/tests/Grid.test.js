import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import theme from '../../../config/theme';

import Grid, { GridItem } from '..';

describe('<A />', () => {
  it('should render properly', () => {
    const { container } = render(
      <Grid theme={theme}>
        <GridItem>Hello</GridItem>
        <GridItem>Grid</GridItem>
      </Grid>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName.toLowerCase()).toEqual('div');
    expect(container.firstChild).toHaveTextContent(/hellogrid/i);
    expect(container.firstChild).toHaveStyleRule('display', 'grid');
  });
});
