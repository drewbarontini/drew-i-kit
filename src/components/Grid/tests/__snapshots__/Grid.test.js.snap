// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should render properly 1`] = `
.c0 {
  display: grid;
}

<div
  class="c0"
>
  <div
    class=""
  >
    Hello
  </div>
  <div
    class=""
  >
    Grid
  </div>
</div>
`;
